index.js

```diff
@@ -218,48 +217,54 @@ let rcl = {
    * @return {Promise<undefined>} A Promise.
    */
   init(context = Context.defaultContext(), argv = process.argv) {
+    
     return new Promise((resolve, reject) => {
+      // check if context has already been initialized
+      if (this._contextToNodeArrayMap.has(context)) {
+        throw new Error('The context has already been initialized.');
+      };
+
       // check argv for correct value and state
       if (!Array.isArray(argv)) {
         throw new TypeError('argv must be an array.');
       }
-      if (argv.reduce((hasNull, arg) => typeof arg !== 'string', false)) {
+      if (argv.reduce((hasNull, arg) => hasNull || typeof arg !== 'string', false)) {
         throw new TypeError('argv elements must not be null');
       }
 
-      let that = this;
-      if (!this._initialized) {
-        getCurrentGeneratorVersion()
-          .then(version => {
-            let forced =
-              version === null ||
-              compareVersions(version, generator.version()) === -1
-                ? true
-                : false;
-            if (forced) {
-              debug(
-                'The generator will begin to create JavaScript code from ROS IDL files...'
-              );
-            }
-
-            generator
-              .generateAll(forced)
-              .then(() => {
-                this._context = context;
-                rclnodejs.init(context.handle(), argv);
-                this._initialized = true;
-                resolve();
-              })
-              .catch(e => {
-                reject(e);
-              });
-          })
-          .catch(e => {
-            reject(e);
-          });
-      } else {
-        throw new Error('The module rclnodejs has been initialized.');
+      // initialize context
+      rclnodejs.init(context.handle(), argv);
+      this._contextToNodeArrayMap.set(context, []);
+
+      if (this._rosVersionChecked) {
```

Comment:
I didn't realize that we could init with more than one context, it's reasonable to add this `if` and resolve the Promise earlier. Thanks for adding the corresponding ut.